File: examples/php/send-push.php
<?php

declare(strict_types=1);

// Create and send a Push Notification campaign (server API, HMAC-signed).
//   PUSHFY_PUSH_KEY=... PUSHFY_PUSH_SECRET=... php send-push.php
//
// HMAC signing is handled by the SDK — you only supply the key/secret pair.

require __DIR__ . '/vendor/autoload.php';

use Pushfy\Pushfy;
use Pushfy\Exception\PushfyException;

$pushfy = new Pushfy([
    'pushKey'    => getenv('PUSHFY_PUSH_KEY') ?: '',
    'pushSecret' => getenv('PUSHFY_PUSH_SECRET') ?: '',
]);

try {
    // 1) Create the campaign.
    $campaign = $pushfy->push->campaigns->create([
        'name'  => 'Weekend promo',
        'title' => 'Sale!',
        'body'  => '50% off — this weekend only',
        'url'   => 'https://example.com/promo',
    ]);

    $id = $campaign['id'] ?? null;
    if ($id === null) {
        fwrite(STDERR, "No campaign id returned.\n");
        exit(1);
    }
    echo "Created campaign #{$id}\n";

    // 2) Send it.
    $pushfy->push->campaigns->send($id);
    echo "Send triggered.\n";

    // 3) Read back delivery metrics.
    $metrics = $pushfy->push->campaigns->metrics($id);
    echo "Metrics:\n";
    echo json_encode($metrics, JSON_PRETTY_PRINT | JSON_UNESCAPED_UNICODE), "\n";
} catch (PushfyException $e) {
    fwrite(STDERR, sprintf(
        "Push campaign failed: status=%d code=%s\n",
        $e->getStatus(),
        (string) $e->getErrorCode()
    ));
    exit(1);
}

File: examples/ruby/send_push.rb
# frozen_string_literal: true

# Create and send a Push Notification campaign with the Pushfy Ruby SDK.
#
#   PUSHFY_PUSH_KEY=... PUSHFY_PUSH_SECRET=... ruby send_push.rb
#
# Env vars:
#   PUSHFY_PUSH_KEY     Push server HMAC key (pushk_...)   (required)
#   PUSHFY_PUSH_SECRET  Push server HMAC secret (pss_...)  (required)
#
# HMAC signing of the V2 endpoints is handled by the SDK.

require "pushfy"

def main
  pushfy = Pushfy::Client.new(
    push_key: ENV.fetch("PUSHFY_PUSH_KEY"),
    push_secret: ENV.fetch("PUSHFY_PUSH_SECRET")
  )

  campaign = pushfy.push.campaigns.create(
    name: "Weekend Promo",
    title: "Sale is live!",
    body: "50% off everything this weekend",
    url: "https://example.com/sale"
  )
  campaign_id = campaign["id"]
  puts "Created campaign #{campaign_id}"

  pushfy.push.campaigns.send(campaign_id)
  puts "Campaign #{campaign_id} sent"

  metrics = pushfy.push.campaigns.metrics(campaign_id)
  puts "Metrics: #{metrics.inspect}"
rescue KeyError => e
  warn "Missing environment variable: #{e.message}"
  exit 1
rescue Pushfy::RateLimitError
  warn "Rate limited — back off and retry."
  exit 1
rescue Pushfy::PushfyError => e
  warn "Failed: #{e.status} #{e.code} #{e.response.inspect}"
  exit 1
end

main
